### PR TITLE
Implement new `Hero Product – Split` pattern

### LINKED ISSUE #9043: Create `Hero product: split` pattern
Split

Check Figma designs F5AAbRJOGbTeEweXPFtHgf-fi-1540_33347 for the latest updates and mobile views.

### PR DESCRIPTION
**Pull request by danielwrobert**
This adds a new pattern for a hero product banner with a 50/50 split, heading and CTA on left and product image on right.

Under the hood, it takes advantage of the Media + Image block, with some preset configurations.

Hero-Split-Design

Fixes #9043 

### Screenshots

CleanShot 2023-04-28 at 15 08 05@2x

### Testing

1. Create a new page or post
2. Make sure the `WooCommerce Hero Product – Split` pattern appears under the WooCommerce category dropdown.
3. Insert in and make sure it shows as expected on the design.

* [ ] Do not include in the Testing Notes 

### WooCommerce Visibility

* [x] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental

### Changelog

> Add the new `WooCommerce Hero Product – Split` pattern.

### CODE REVIEW

File: patterns/hero-product-split.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
+<?php
+/**
+ * Title: WooCommerce Hero Product - Split
```
Comment (albarin) [inline patterns/hero-product-split.php]: I think you can remove the `WooCommerce` prefix here. We decided to remove it and all the patterns we are merging don't have it.

File: patterns/hero-product-split.php [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+			<div class="wp-block-buttons" style="margin-bottom:var(--wp--preset--spacing--40)">
+				<!-- wp:button -->
+				<div class="wp-block-button"><a class="wp-block-button__link wp-element-button">Shop now</a></div>
+				<!-- /wp:button -->
+			</div>
+			<!-- /wp:buttons -->
+		</div>
+
+		<figure class="wp-block-media-text__media">
+			<img src="https://images.unsplash.com/photo-1610873397492-e29c42344c00?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1277&q=80" alt="Woman in red, black, and white plaid hoodie." />
```
Comment (albarin) [inline patterns/hero-product-split.php]: I'm not sure which is the right approach here, just wondering if we want to link to external images or include them on the plugin like the approach I took here https://github.com/woocommerce/woocommerce-blocks/pull/9180/files ? 🤔

**Review by albarin — COMMENTED** [review]
The code looks good, but this is what I'm getting 👇  in the preview looks good, but not in the editor.
Screenshot 2023-05-02 at 09 45 18

File: patterns/hero-product-split.php (hunk 1 above)
Comment (danielwrobert) [reply patterns/hero-product-split.php]: Updated in 2f38c71.

File: patterns/hero-product-split.php (hunk 2 above)
Comment (danielwrobert) [reply patterns/hero-product-split.php]: Thanks! Updated in c9d9e71.

**Comment by danielwrobert** [comment]
> ... in the preview looks good, but not in the editor.

I think the group wrapper was constraining the block when added to a Post or Page. This should be fixed in cb721d0.

**Review by albarin — APPROVED** [review]

**Pull request merged**